File: gluecodium/src/test/resources/smoke/lambdas/output/android/com/example/smoke/StandaloneProducerImpl.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
class StandaloneProducerImpl extends NativeBase implements StandaloneProducer {
    protected StandaloneProducerImpl(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native String apply();
}

File: gluecodium/src/test/resources/smoke/lambdas/output/android-kotlin/com/example/smoke/StandaloneProducerImpl.kt
/*

 *
 */

@file:JvmName("StandaloneProducerImpl")

package com.example.smoke

import com.example.NativeBase

/**
 * @suppress
 */
class StandaloneProducerImpl : NativeBase, StandaloneProducer {
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}



    override external fun apply() : String



    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
